src/communicate.rs

```diff
@@ -88,12 +93,15 @@ pub fn try_write_to_peer(peer: Peer, msg: WireMsg, token: Token) -> R<()> {
 
 /// This will fail if we don't have a connection to the peer or if the peer is in an invalid state
 /// to be sent a message to.
-pub fn write_to_peer(peer_addr: SocketAddr, msg: WireMsg, token: Token) -> R<()> {
+pub fn write_to_peer(peer: Peer, msg: WireMsg, token: Token) -> R<()> {
```

Comment:
Should this not be private and called from `write_to_peer_connection` So we always need to pass socketaddr, but if we are already connected then this method runs, otherwise we create a connection.

Reply:
Could be, but this doesn't seem to be related to the purpose of this PR, so I'd prefer to not do it here.